tests/dt-posts/dt-posts/unit-test-post-comments.php

```diff
@@ -0,0 +1,99 @@
+<?php
+require_once( get_template_directory() . '/tests/dt-posts/tests-setup.php' );
+
+class DT_Posts_DT_Posts_Post_Comments extends WP_UnitTestCase {
+
+    public static $sample_contact = [
+        'title'           => 'Bob',
+        'overall_status'  => 'active',
+        'milestones'      => [
+            "values" => [
+                [ "value" => 'milestone_has_bible' ],
+                [ "value" => "milestone_baptizing" ]
+            ]
+        ],
+        'baptism_date'    => "2022-01-13",
+        "location_grid"   => [ "values" => [ [ "value" => '100089589' ] ] ],
+        "assigned_to"     => "1",
+        "requires_update" => true,
+        "nickname"        => "Bob the builder",
+        "contact_phone"   => [ "values" => [ [ "value" => "798456780" ] ] ],
+        "contact_email"   => [ "values" => [ [ "value" => "bob@example.com" ] ] ],
+        "tags"            => [ "values" => [ [ "value" => "tag1" ], [ "value" => "tagToDelete" ] ] ],
+    ];
+    public static $contact = null;
+
+    public static function setupBeforeClass() {
+        //setup custom fields for each field type and custom tile.
+        $user_id = wp_create_user( "dispatcher1", "test", "test2@example.com" );
+        wp_set_current_user( $user_id );
+        $current_user = wp_get_current_user();
+        $current_user->set_role( 'dispatcher' );
+
+        self::$contact = DT_Posts::create_post( "contacts", self::$sample_contact, true, false );
+    }
+
+    public function test_comment_creation_with_valid_data() {
+
+        $comment_args = [
+            'comment_date' => '2022-01-13 12:00:00'
+        ];
+        $comment_html = 'Valid data comments test';
+        $comment_id   = DT_Posts::add_post_comment( self::$contact['post_type'], self::$contact['ID'], $comment_html, 'comment', $comment_args, false, true );
+        $comments     = DT_Posts::get_post_comments( self::$contact['post_type'], self::$contact['ID'], false );
+
+        $this->assertNotWPError( $comment_id );
+        $this->assertSame( 1, (int) $comments['total'] );
+        $this->assertSame( $comment_html, $comments['comments'][0]['comment_content'] );
+    }
+
+    public function test_comment_creation_with_invalid_data() {
+
+        $comment_args  = [
+            'comment_date' => '0000-00-00 00:00:00'
```

Comment:
@kodinkat , let's test a couple more invalid? cases. I can think of
- '' (empty string)
- null,
- 7/6/2009 14:16

Reply:
....however, any string literal; i.e. `null`; which does not adhere with required date format, will trigger exception...